Training4/thongnq-training4/src/App.js

```diff
@@ -0,0 +1,34 @@
+import React, {useState, useEffect, useCallback} from "react";
+
+import axios from "axios";
+
+import Country from "./task2/Country";
+
+import debounce from 'lodash.debounce';
+
+const countriesAPI = "https://restcountries.eu/rest/v2";
+
+const App = () => {
+  const [countries, setCountries] = useState([]);
+
+  const hanldeLoadCountry = (api) => {
+    axios.get(api).then(response => {
+      setCountries(response.data);
+    })
+  }
+
+  const debounceOnChange = useCallback(debounce(hanldeLoadCountry, 1000), []);
+
+  useEffect(() => {
+    hanldeLoadCountry(countriesAPI + "/all");
+  }, [])
+
+  return (
+    <div className="App">
+      <input type="text" onChange={e => e.target.value ? debounceOnChange(countriesAPI + "/name/" + e.target.value) : debounceOnChange(countriesAPI + "/all")}/>
```

Comment:
@Thong-Nguyen-Zigvy please try to use the controlled component approach for this case